File: back-end/src/main/java/com/trainingProject/moviesWebApp/service/UserService.java
package com.trainingProject.moviesWebApp.service;

import com.trainingProject.moviesWebApp.dto.UserDto;
import com.trainingProject.moviesWebApp.dto.UserLoginDto;

public interface UserService {
    boolean authenticate (UserLoginDto userLoginDto);
    UserDto register(UserDto userDto);
}

File: frontEnd/src/app/_services/user.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { User } from '@app/_models/user';
import { UserDto } from '@app/dto/userDto';
import { Observable } from 'rxjs';

@Injectable({ providedIn: 'root' })
export class UserService {

    private baseUrl = 'http://localhost:8080/users';

    constructor(private http: HttpClient) { }

    public register(userDto: UserDto): Observable<User> {
        return this.http.post<User>(`${this.baseUrl}/register`, userDto);
    }
}
